### ISSUE TITLE
[Bug]: Recently visited pages don't show up in URL bar search

### ISSUE THREAD

**Issue by buttercookie42** [mozilla-mobile/fenix#21535]
### Steps to reproduce

1. Browser around some pages, e.g. I visited https://www.buttercookie.de/The%20Burkiss%20Way/Transcripts/ and then a few of the individual transcripts.
2. Enter some text related to those pages (e.g in the above example "Burkiss") in the URL bar.

### Expected behaviour

All those pages you visited appear in the search results.

### Actual behaviour

I get *some* history results from what must presumably an earlier occasion where I visited a few of those pages, i.e. the history results aren't *completely* empty, but on the other hand *none* of the additional pages that I know I visited right now show up.
They *do* all appear in the dedicated history view, though.

Basically just like https://github.com/mozilla-mobile/fenix/issues/17831, except I'm seeing this right now with a current Nightly.

### Device name

Moto G4 Play

### Android version

Android 6.0.1

### Firefox release type

Firefox Nightly

### Firefox version

94.0a1 (Build #2015836489)

**Comment by DiegoARL38** [comment]
This happens to me, as well. Also, search suggestions from old history entries are quite inconsistent.

It started on Sunday's builds, at least for me. Don't know about Friday's or Saturday's, because I suffered from a crash related to history metadata. I'd assume either whatever introduced #21506  or its resolution is the culprit.

**Comment by DiegoARL38** [comment]
I took a look at the Android Components repository and, judging by recent activity, this is something being actively worked on. I'm not a programmer, so I can't discern more...

**Comment by buttercookie42** [comment]
Giving it a quick test, this does seem to be working normally again.
